FDApy/representation/functional_data.py

```diff
@@ -0,0 +1,527 @@
+    if len(set(lengths)) > 1:
+        raise ValueError("The number of observations is different across the"
+                         " dimensions.""")
+    return None
+
+###############################################################################
+# Class FunctionalData
+
+
+class FunctionalData(ABC):
+    """Metaclass for the definition of diverse functional data objects.
+
+    Parameters
+    ----------
+    argvals: list
+    values: list
+    category: str, {'univariate', 'irregular', 'multivariate'}
+    """
+
+    @staticmethod
+    @abstractmethod
+    def _check_argvals(argvals):
+        _check_type(argvals, dict)
+
+    @staticmethod
+    @abstractmethod
+    def _check_values(values):
+        pass
+
+    @staticmethod
+    @abstractmethod
+    def _check_argvals_values(argvals, values):
+        pass
+
+    def __init__(self, argvals, values, category):
+        """Initialize FunctionalData object."""
+        super().__init__()
+        self.argvals = argvals
+        self.values = values
+        self.category = category
+
+    def __repr__(self):
+        """Override print function."""
+        return (f"{self.category.capitalize()} functional data object with"
+                f" {self.n_obs} observations on a {self.n_dim}-dimensional"
+                " support.")
+
+    @abstractmethod
+    def __getitem__(self, index):
+        """Function call when self[index]."""
+        pass
+
+    def __add__(self, obj):
+        """Override add function."""
+        pass
+
+    def __sub__(self, obj):
+        """Override sub function."""
+        pass
+
+    def __mul__(self, obj):
+        """Overrude mul function."""
+        pass
+
+    def __rmul__(self, obj):
+        """Override rmul function."""
+        return self * obj
+
+    @property
+    def argvals(self):
+        """Getter for argvals."""
+        return self._argvals
+
+    @argvals.setter
+    @abstractmethod
+    def argvals(self, new_argvals):
+        pass
+
+    @property
+    def argvals_stand(self):
+        """Getter for argvals_stand."""
+        return self._argvals_stand
+
+    @property
+    def values(self):
+        """Getter for values."""
+        return self._values
+
+    @values.setter
+    @abstractmethod
+    def values(self, new_values):
+        pass
+
+    @property
+    def category(self):
+        """Getter for category."""
+        return self._category
+
+    @category.setter
+    def category(self, new_category):
+        self._category = new_category
+
+    @property
+    def n_obs(self):
+        """Number of observations of the functional data.
+
+        Returns
+        -------
+        n_obs: int
+            Number of observations within the functional data.
+
+        """
+        return len(self.values)
+
+    @property
+    @abstractmethod
+    def range_obs(self):
+        pass
+
+    @property
+    def n_dim(self):
+        """Number of input dimension of the functional data.
```

Comment:
Codacy found an issue: [First line should be in imperative mood; try rephrasing (found 'Number') (D401)](https://app.codacy.com/manual/StevenGolovkine/FDApy/pullRequest?prid=5929506)